src/libs/blueprint/conduit_blueprint_mesh_examples.cpp

```diff
@@ -3421,6 +3421,165 @@ adjset_uniform(Node &res)
+        
+        // side where x = 1                     // side where x = -1
+        sub_connec[i * 60 + 8] = 0 + i * 20;    sub_connec[i * 60 + 12] = 2 + i * 20;
+        sub_connec[i * 60 + 9] = 1 + i * 20;    sub_connec[i * 60 + 13] = 3 + i * 20;
+        sub_connec[i * 60 + 10] = 5 + i * 20;   sub_connec[i * 60 + 14] = 7 + i * 20;
+        sub_connec[i * 60 + 11] = 4 + i * 20;   sub_connec[i * 60 + 15] = 6 + i * 20;
+        
+        // side where z = 1                     // side where z = -1
+        sub_connec[i * 60 + 16] = 0 + i * 20;   sub_connec[i * 60 + 20] = 1 + i * 20;
+        sub_connec[i * 60 + 17] = 3 + i * 20;   sub_connec[i * 60 + 21] = 2 + i * 20;
+        sub_connec[i * 60 + 18] = 7 + i * 20;   sub_connec[i * 60 + 22] = 6 + i * 20;
+        sub_connec[i * 60 + 19] = 4 + i * 20;   sub_connec[i * 60 + 23] = 5 + i * 20;
+
+        // PRISM 1
+        sub_connec[i * 60 + 24] = 9 + i * 20;  sub_connec[i * 60 + 28] = 8 + i * 20;  sub_connec[i * 60 + 32] = 8 + i * 20;
+        sub_connec[i * 60 + 25] = 10 + i * 20; sub_connec[i * 60 + 29] = 9 + i * 20;  sub_connec[i * 60 + 33] = 10 + i * 20;
+        sub_connec[i * 60 + 26] = 13 + i * 20; sub_connec[i * 60 + 30] = 12 + i * 20; sub_connec[i * 60 + 34] = 13 + i * 20;
+        sub_connec[i * 60 + 27] = 12 + i * 20; sub_connec[i * 60 + 31] = 11 + i * 20; sub_connec[i * 60 + 35] = 11 + i * 20;
+
+        sub_connec[i * 60 + 36] = 8 + i * 20;   sub_connec[i * 60 + 39] = 11 + i * 20;
+        sub_connec[i * 60 + 37] = 9 + i * 20;   sub_connec[i * 60 + 40] = 12 + i * 20;
+        sub_connec[i * 60 + 38] = 10 + i * 20;  sub_connec[i * 60 + 41] = 13 + i * 20;
+
+        // PRISM 2
+        sub_connec[i * 60 + 42] = 15 + i * 20; sub_connec[i * 60 + 46] = 14 + i * 20; sub_connec[i * 60 + 50] = 14 + i * 20;
+        sub_connec[i * 60 + 43] = 16 + i * 20; sub_connec[i * 60 + 47] = 15 + i * 20; sub_connec[i * 60 + 51] = 16 + i * 20;
+        sub_connec[i * 60 + 44] = 19 + i * 20; sub_connec[i * 60 + 48] = 18 + i * 20; sub_connec[i * 60 + 52] = 19 + i * 20;
+        sub_connec[i * 60 + 45] = 18 + i * 20; sub_connec[i * 60 + 49] = 17 + i * 20; sub_connec[i * 60 + 53] = 17 + i * 20;
+
+        sub_connec[i * 60 + 54] = 14 + i * 20; sub_connec[i * 60 + 57] = 17 + i * 20;
+        sub_connec[i * 60 + 55] = 15 + i * 20; sub_connec[i * 60 + 58] = 18 + i * 20;
+        sub_connec[i * 60 + 56] = 16 + i * 20; sub_connec[i * 60 + 59] = 19 + i * 20;
+    }
+
+    chain_topo["subelements/sizes"].set(conduit::DataType::int64(howmany * 16));
+    int64* sub_sizes = chain_topo["subelements/sizes"].value();
+    for (int i = 0; i < howmany * 16; i ++)
+    {
+        // this ensures sizes will be of the form {4,4,4,4,4,4,4,4,4,3,3,4,4,4,3,3, 4,4,4,4,4,4,4,4,4,3,3,4,4,4,3,3, ...}
+        int imod16 = i % 16;
+        sub_sizes[i] = ((imod16 < 9) || ((imod16 > 10) && (imod16 < 14))) ? 4 : 3;
+    }
+
+    chain_topo["subelements/offsets"].set(conduit::DataType::int64(howmany * 16));
```

Comment:
This code can also be eliminated by using [`blueprint::mesh::topology::unstructured::generate_offsets`](https://github.com/LLNL/conduit/blob/develop/src/libs/blueprint/conduit_blueprint_mesh.hpp#L362). In fact, I think just a call here will do; it looks like the function handles both the `elements/offsets` and `subelements/offsets` paths.

Reply:
would it be as simple as this?
```blueprint::mesh::topology::unstructured::generate_offsets(chain_topo, chain_topo["elements/offsets"]);```
???

Reply:
Yes, that's correct!